### PR TITLE
Move to standard venv with lxplus Python3.11 installation and to awkward v2

### PR DESCRIPTION
**Pull request by danielhundhausen**
Now Python3.11 is available by default on LXPLUS and we can use this instead of having to install conda.
Advantages:
- Easier setup process as (mini)conda installation is not necessary
- More flexibility regarding dependencies, since conda struggles to resolve the dependency tree

Awkward was updated from version 1 to 2 in this PR. I have tested the caching and turnon plotting scritps which appear to work fine with these modifications. Two changes in awkward were accounted for:
1. `to_parquet` changed the `where` kwarg to `destination`
2. slightly different behaviour of `ak.cartesian` leads to one less dimension in output and no inner zeros (which we previously removed in a separate function)

### CODE REVIEW

File: objectPerformance/src/turnon_collection.py [hunk 1]
```diff
@@ -142,8 +142,8 @@ def _match_test_to_ref(self):
             pt_max = ak.argmax(ref_test["test"]["pt"][pass_dR], axis=-1,
                                keepdims=True)
             if ("iso" not in suffix):
-                self.numerators["ref"][test_obj] = ref_test["ref"][suffix][pass_dR][pt_max][:, :, 0]  # noqa
-            self.numerators["test"][test_obj] = ref_test["test"][suffix][pass_dR][pt_max][:, :, 0]  # noqa
+                self.numerators["ref"][test_obj] = ref_test["ref"][suffix][pass_dR][pt_max][:, 0]  # noqa
+            self.numerators["test"][test_obj] = ref_test["test"][suffix][pass_dR][pt_max][:, 0]  # noqa
```
Comment (bonanomi) [inline objectPerformance/src/turnon_collection.py:146]: Have you tested this? Does the modification alter the plots? Why did we have three levels of nestedness before?

File: objectPerformance/src/turnon_collection.py [hunk 2]
```diff
@@ -325,23 +325,21 @@ def _skim_to_hists(self):
                 bins=self.bins
             )
 
-    def _remove_inner_nones_zeros(self, arr):
+    def _remove_inner_nones(self, arr):
         sel_arr_not_none = ~ak.is_none(arr, axis=-1)
-        sel_arr_not_zero = ak.num(arr) > 0
```
Comment (bonanomi) [inline objectPerformance/src/turnon_collection.py:330]: Are you sure this is not needed? We should at least check that the object performance does not change anytime we include this kind of modification.

File: objectPerformance/src/turnon_collection.py (hunk 1 above)
Comment (danielhundhausen) [reply objectPerformance/src/turnon_collection.py:146]: Yes, I have tested it on a few objectPerformance configs and the plots seem to be reproduced this way. But tbh I have not understood from first principles what exactly the change here is.

File: objectPerformance/src/turnon_collection.py (hunk 2 above)
Comment (danielhundhausen) [reply objectPerformance/src/turnon_collection.py:330]: I think this is related to the other change you commented on above. I will try to find some change log where we can see what the cause of this change is. The objectPerformance plots are reproduced with these changes and the new awkward version.

File: objectPerformance/src/turnon_collection.py (hunk 1 above)
Comment (artlbv) [reply objectPerformance/src/turnon_collection.py:146]: could it be due to the awkward version change?
btw, there is a dedicated function to the get the first element in the event: [`ak.firsts`](https://awkward-array.org/doc/main/reference/generated/ak.firsts.html) which also fills the empty rows with `None`s. Should we switch to that instead?

**Comment by artlbv** [comment]
> slightly different behaviour of ak.cartesian leads to one less dimension in output and no inner zeros (which we previously removed in a separate function)

can this be due to the `nested` argument? maybe the default changed (though we seem to always set nested to `True`)

File: objectPerformance/src/turnon_collection.py (hunk 1 above)
Comment (danielhundhausen) [reply objectPerformance/src/turnon_collection.py:146]: I tested `ak.firsts` and indeed it produces the same result judging from the electron iso plots [here](https://dhundhausen.web.cern.ch/L1_PhaseII/ak_firsts/) with `ak.firsts` and in the directory above that with the slicing.

File: objectPerformance/src/turnon_collection.py (hunk 2 above)
Comment (danielhundhausen) [reply objectPerformance/src/turnon_collection.py:330]: I researched a bit but was not able to find anything. But evidently there was a change. What do you suggest moving forward?

**Comment by danielhundhausen** [comment]
UPDATE: apparently there is [a bug](https://github.com/scikit-hep/awkward/issues/2928#issuecomment-1884839092) in `ak.cartesian` so I rolled back the changes to accommodate the different behaviour in `turnon_collections.py`. We should probably wait with merging this until the bug is fixed.

**Comment by danielhundhausen** [comment]
All plots can be reproduced with the update included. [Here](https://dhundhausen.web.cern.ch/L1_PhaseII/comparisons-version-update/) are various comparisons of plots. There were some deviations from the archived plots from Artur which turn out to disappear when reproducing everything with the old setup at the current version of the code on main.

**Pull request merged**